src/Symfony/Component/Form/Tests/Extension/Validator/DataTransformer/ViewObjectTransformer.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace Symfony\Component\Form\Tests\Extension\Validator\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+* Transforms Object to array for view
+*
+* @author Charles J. C. Elling <charles@denumeris.com>
+*/
+class ViewObjectTransformer implements DataTransformerInterface {
```

Comment:
This transformer is for testing the correct mapping of validation violations when using a _data transformer_, i could implemente a simpler version considering only public properties.

The most common case for implementing a custom data transformer is using embedded forms synchronised to data entities that need to use the _id_ of the entity to decide whether to retrieve or create the data object before setting the property values.